astropy_helpers/tests/test_openmp_helpers.py

```diff
@@ -0,0 +1,38 @@
+import os
+import sys
+from copy import deepcopy
+from distutils.core import Extension
+
+from ..openmp_helpers import add_openmp_flags_if_available
+from ..setup_helpers import _module_state, register_commands
+
+IS_TRAVIS_LINUX = os.environ.get('TRAVIS_OS_NAME', None) == 'linux'
+IS_APPVEYOR = os.environ.get('APPVEYOR', None) == 'True'
+PY3_LT_35 = sys.version_info[0] == 3 and sys.version_info[1] < 5
+
+_state = None
+
+
+def setup_function(function):
+    global state
+    state = deepcopy(_module_state)
+
+
+def teardown_function(function):
+    _module_state.clear()
+    _module_state.update(state)
+
+
+def test_add_openmp_flags_if_available():
+
+    register_commands('openmp_testing', '0.0', False)
+
+    using_openmp = add_openmp_flags_if_available(Extension('test', []))
+
+    # Make sure that on Travis (Linux) and AppVeyor OpenMP does get used (for
+    # MacOS X usually it will not work but this will depend on the compiler).
+    # Having this is useful because we'll find out if OpenMP no longer works
+    # for any reason on platforms on which it does work at the time of writing.
+    # OpenMP doesn't work on Python 3.x where x<5 on AppVeyor though.
+    if IS_TRAVIS_LINUX or (IS_APPVEYOR and not PY3_LT_35):
+        assert using_openmp
```

Comment:
I would think you should invert this test so that it defaults to running if I run e.g. locally.  That is, something like:

```
if not IS_TRAVIS_OSX and not (PY3_LT_35 and IS_APPVETOR)
    assert using_openmp
```

Reply:
The issue with that is that it is not guaranteed to pass locally, e.g. if you run on MacOS X with the Apple clang then it won't pass. The idea of the current logic is that it only checks that OpenMP is being used on platforms/circumstances where we *know* it will work. What do you think?

Reply:
It still concerns me a bit there's no way to check this locally, but now that I realized it runs regardless and it's only the assert that's being checked here, I think it's ok. So all set and will merge